specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/QueryPerformanceInsights.json

```diff
@@ -0,0 +1,684 @@
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/TopQueryStatisticsInput"
+            },
+            "description": "The required parameters for retrieving top query statistics."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TopQueryStatisticsResultList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMariaDB/servers/{serverName}/waitStatistics/{waitStatisticsId}": {
+      "get": {
+        "tags": [
+          "WaitStatistics"
+        ],
+        "operationId": "WaitStatistics_Get",
+        "x-ms-examples": {
+          "WaitStatisticsGet": {
+            "$ref": "./examples/WaitStatisticsGet.json"
+          }
+        },
+        "description": "Retrieve wait statistics for specified identifier.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WaitStatisticsIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/WaitStatistic"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMariaDB/servers/{serverName}/waitStatistics": {
+      "get": {
+        "tags": [
+          "WaitStatistics"
+        ],
+        "operationId": "WaitStatistics_ListByServer",
+        "x-ms-examples": {
+          "WaitStatisticsListByServer": {
+            "$ref": "./examples/WaitStatisticsListByServer.json"
+          }
+        },
+        "description": "Retrieve wait statistics for specified aggregation window.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "parameters",
```

Comment:
GET cannot have a body

Reply:
So, these are parameters in the query. How should they be annotated?


**/topQueryStatistics** API
`
https://management.southcentralus1-c.control.orcasql-scus1-c.mscds.com/modules/ArmMySQL/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/mysql_resource/providers/Microsoft.DBforMySQL/servers/test-server/topQueryStatistics?api-version=2018-06-01-privatepreview&numberOfTopQueries=5&aggregationFunction=avg&observedMetric=duration&observationStartTime=2019-04-01T20:00:00.000Z&observationEndTime=2019-05-13T20:15:00.000Z&aggregationWindow=PT15M
`
* numberOfTopQueries
* aggregationFunction
* observedMetric
* observationStartTime
* observationEndTime


**/waitStatistics** API
`
https://management.southcentralus1-c.control.orcasql-scus1-c.mscds.com/modules/ArmMySQL/subscriptions/ffffffff-ffff-ffff-ffff-ffffffffffff/resourceGroups/mysql_resource/providers/Microsoft.DBforMySQL/servers/test-server/waitStatistics?api-version=2018-06-01-privatepreview&observationStartTime=2019-05-05T23:30:00.000Z&observationEndTime=2019-05-06T23:30:00.000Z&aggregationWindow=PT15M
`
* observationStartTime
* observationEndTime
* aggregationWindow

Reply:
Are these results generated at API call time?  These seem more like query APIs than resource collection GETs.  How do the individual GETs work if the results are generated?

If it is generating results I would suggest using a POST action and either representing your parameters in [ODATA compliant query parameters](https://www.odata.org/documentation/odata-version-3-0/url-conventions/) if possible or as a request body if not.

Reply:
These results are _not_ generated at API call time. They are continuously aggregated and persisted on the system tables as a background process while the feature is enabled. The APIs here retrieves the rows from the respective system table.